sbi/inference/base.py

```diff
@@ -152,6 +153,42 @@ def __init__(
             median_observation_distances=[], epochs=[], best_validation_log_probs=[],
         )
 
+    def _append_training_data(
+        self,
+        theta: Tensor,
+        x: Tensor,
+        external_data: Tuple[Tensor, Tensor],
+        round: int,
+        exclude_invalid_x: bool,
+    ) -> None:
+        """
+        Take simulation data and external data, and merge them into attributes.
+
+        Args:
+            theta: Simulation parameters.
+            x: Simulation outputs.
+            external_data: Externally provided data consisting of (theta,x).
+            round: Current round.
+            exclude_invalid_x: Whether to exclude simulation outputs `x=NaN` or `x=±∞`
+                during training. Expect errors, silent or explicit, when `False`.
+        """
+
+        # In the first round, the user can externally provide data.
+        if external_data is not None and round == 0:
+            theta = torch.cat((external_data[0], theta))
+            x = torch.cat((external_data[1], x))
```

Comment:
Would remove this and introduce a separate method, e.g., `_append_external_data`. The advantages are that 1) the method can do extra checks on the external data provided (i.e., is it in a valid format), 2) `_append_training_data` keeps a clean signature, always requiring `theta` and `x`